client/src/components/ElementEditor/tests/Header-test.js

```diff
@@ -1,307 +1,217 @@
-      const wrapper = shallow(
-        <Header
-          element={element}
-          areaId={1}
-          type={type}
-          expandable
-          previewExpanded
-          ElementActionsComponent={ElementActionsComponent}
-          connectDragSource={content => content}
-          connectDragPreview={content => content}
-          onDragEnd={() => {}}
-        />
-      );
-
-      const expandButton = wrapper.find('.element-editor-header__expand');
-      expect(expandButton.length).toBe(1);
-      expect(expandButton.hasClass('font-icon-up-open-big')).toBe(true);
-    });
+test('Header should render the icon', () => {
+  const { container } = render(<Header {...makeProps()}/>);
+  expect(container.querySelector('i.font-icon-block-file')).not.toBeNull();
+});
 
-    it('should not render a "caret" button for a broken element', () => {
-      const wrapper = shallow(
-        <Header
-          element={element}
-          areaId={1}
-          type={typeBroken}
-          expandable={false}
-          ElementActionsComponent={ElementActionsComponent}
-          connectDragSource={content => content}
-          connectDragPreview={content => content}
-          onDragEnd={() => {}}
-        />
-      );
+test('Header should render the title', () => {
+  const { container } = render(
+    <Header {...makeProps({
+      element: {
+        id: '12',
+        title: 'Sample File Block'
+      }
+    })}
+    />
+  );
+  expect(container.querySelector('.element-editor-header__title').textContent).toBe('Sample File Block');
+});
 
-      const expandButton = wrapper.find('.element-editor-header__expand');
-      expect(expandButton.length).toBe(0);
-    });
+test('Header should render the title for broken elements', () => {
+  const { container } = render(
+    <Header {...makeProps({
+      type: {
+        broken: true,
+        obsoleteClassName: 'RemovedClass'
+      }
+    })}
+    />
+  );
+  expect(container.querySelector('.element-editor-header__title').textContent).toBe('This element is of obsolete type RemovedClass.');
+});
 
-    it('should render an ElementActions component when the element is expandable', () => {
-      const wrapper = shallow(
-        <Header
-          element={element}
-          areaId={1}
-          type={type}
-          expandable
-          ElementActionsComponent={ElementActionsComponent}
-          connectDragSource={content => content}
-          connectDragPreview={content => content}
-          onDragEnd={() => {}}
-        />
-      );
+test('Header should contain a Tooltip', async () => {
+  const { container } = render(
+    <Header {...makeProps({
+      element: {
+        id: '13',
+        title: 'Sample File Block',
+      }
+    })}
+    />
+  );
+  fireEvent.mouseOver(container.querySelector('#element-icon-13.font-icon-block-file'));
+  const tooltip = await screen.findByText('File');
+  expect(tooltip.getAttribute('role')).toBe('tooltip');
```

Comment:
This validates whether a tooltip will appear - but the original test (and for the one below) are checking whether a tooltip exists _at all_ - might be worth doing both, i.e. validate that the element exists in the DOM at all, and then check whether it appears on hover.
This will especially make the one below more robust, I think.

Reply:
Not doing